src/fty_common_messagebus_malamute.cc

```diff
@@ -329,19 +332,25 @@ namespace messagebus {
     {
         log_debug ("%s - received mailbox message from '%s' subject '%s'", m_clientName.c_str(), from, subject);
 
-        Message msg = _fromZmsg(message);
+        Message msg = _fromZmsg(message, from);
 
         bool syncResponse = false;
         if( m_syncUuid != "" ) {
+            std::string uuid_read;
             auto it = msg.metaData().find(Message::CORRELATION_ID);
             if( it != msg.metaData().end() ) {
-                if( m_syncUuid == it->second ) {
-                    std::unique_lock<std::mutex> lock(m_cv_mtx);
-                    m_syncResponse = msg;
-                    m_cv.notify_one();
-                    m_syncUuid = "";
-                    syncResponse = true;
-                }
+                uuid_read = it->second;
+            }
+            else {
+                // Get uuid in fist part of message (for legacy message without meta data)
```

Comment:
We shouldn't implement any protocol here. The client will have to manage the protocol his self. UUID is part of the protocol.

I would prefer that if the request is RAW format, we transfer the first reply in case of sync.

Reply:
Take into account